sklearn/linear_model/_glm/glm.py

```diff
@@ -80,18 +77,33 @@ class GeneralizedLinearRegressor(RegressorMixin, BaseEstimator):
         added to the linear predictor (X @ coef + intercept).
 
     family : {'normal', 'poisson', 'gamma', 'inverse-gaussian'} \
-            or an ExponentialDispersionModel instance, default='normal'
+            or a BaseLoss instance, default='normal'
         The distributional assumption of the GLM, i.e. which distribution from
         the EDM, specifies the loss function to be minimized.
 
-    link : {'auto', 'identity', 'log'} or an instance of class BaseLink, \
-            default='auto'
-        The link function of the GLM, i.e. mapping from linear predictor
-        `X @ coeff + intercept` to prediction `y_pred`. Option 'auto' sets
-        the link depending on the chosen family as follows:
+        .. deprecated:: 1.1
+           `family` is deprecated in 1.1 and will be removed in 1.3.
+           Use `_base_loss` instead.
```

Comment:
```suggestion
           Use the private `_base_loss` instead.
```

Reply:
Actually I am not sure we need a deprecation warning here since the `GeneralizedLinearRegressor` class is private (it was not exposed under `sklearn.linear_model`) and the family constructor parameter is not exposed in the public concrete descendant classes.